### DISCUSSION TITLE
How to create ZIP files including the trailing descriptor signatures

### DISCUSSION THREAD

**Discussion by simonbuchan** [gildas-lormeau/zip.js#540] [Q&A] [answered]
As described in APPNOTE.TXT:

> 4.3.9.3 Although not originally assigned a signature, the value
>         0x08074b50 has commonly been adopted as a signature value
>         for the data descriptor record.  Implementers SHOULD be
>         aware that ZIP files MAY be encountered with or without this
>         signature marking data descriptors and SHOULD account for
>         either case when reading ZIP files to ensure compatibility.

In other words, the local entry for such files looks like:

```
[header...] [data...] "PK\7\8" crc32  compressed_size uncompressed_size
```

The AWS CloudFormation resource schema is (currently?) an example of such a file: https://schema.cloudformation.us-east-1.amazonaws.com/CloudformationSchema.zip

**Comment by gildas-lormeau** [comment, marked as answer]
You have to set the option `dataDescriptorSignature` to `true`, see https://gildas-lormeau.github.io/zip.js/api/interfaces/ZipWriterConstructorOptions.html#dataDescriptorSignature.

**Comment by simonbuchan** [comment]
Gah. This totally does work, sorry for the confusion - I somehow managed to get an error earlier, but I can't figure out what I was doing wrong now.

**Reply by gildas-lormeau** [reply]
No problem, it was actually set to `true` by default in the past but macOS did not support it.

**Comment by simonbuchan** [comment]
(to be clear, I was meaning to read such files, but this is handy anyway!)
